src/layout.js

```diff
@@ -204,6 +204,18 @@ export function getLengthUnits(length) {
   return m[0];
 }
 
+/**
+ * Returns units from the CSS length value
+ * (including percent unit).
+ * @param {!LengthDef} length
+ * @return {string}
+ */
+export function getLengthUnitsOrPercent(length) {
+  assertLengthOrPercent(length);
+  const m = user().assert(length.match(/[a-z%]+/i),
```

Comment:
We should probably match the assertion regex in `assertLengthOrPercent` instead of `a-z`.